sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/V1Writes.scala

```diff
@@ -202,10 +202,24 @@ object V1WritesUtils {
   def isOrderingMatched(
       requiredOrdering: Seq[Expression],
       outputOrdering: Seq[SortOrder]): Boolean = {
-    if (requiredOrdering.length > outputOrdering.length) {
+    val (constantOutputOrdering, nonConstantOutputOrdering) = outputOrdering.partition {
+      case SortOrder(child, _, _, _, isConstant) => isConstant || child.foldable
+    }
+
+    val effectiveRequiredOrdering = requiredOrdering.filterNot { requiredOrder =>
+      constantOutputOrdering.exists {
+        case s @ SortOrder(alias: Alias, _, _, _, true) =>
```

Comment:
Seems like if we don't use the whole `Alias` in `SortOrder` then we can simplify this logic.